tuned/daemon/controller.py

```diff
@@ -417,3 +417,74 @@ def instance_get_devices(self, instance_name, caller = None):
 		rets = "Instance '%s' not found" % instance_name
 		log.error(rets)
 		return (False, rets, [])
+
+	@exports.export("ssa{ss}", "bs")
+	def instance_create(self, plugin_name, instance_name, options, caller = None):
+		"""Dynamically create a plugin instance
+
+		Parameters:
+		plugin_name -- name of the plugin
+		instance_name -- name of the new instance
+		dict of string-string -- options for the new instance
+
+		Return:
+		bool -- True on success
+		string -- error message or "OK"
+		"""
+		plugins = {p.name: p for p in self._daemon._unit_manager.plugins}
+		if not plugin_name in plugins.keys():
+			rets = "Plugin '%s' not found" % plugin_name
+			log.error(rets)
+			return (False, rets)
+		plugin = plugins[plugin_name]
+		instance = plugin.create_instance(instance_name, None, None, None, None, options)
+		plugin.initialize_instance(instance)
+		instance._created_dynamically = True
+		self._daemon._unit_manager.instances.append(instance)
+		log.info("Created dynamic instance '%s' of plugin '%s'" % (instance_name, plugin_name))
+		return (True, "OK")
+
+	@exports.export("ss", "bs")
+	def instance_destroy(self, plugin_name, instance_name, caller = None):
```

Comment:
Maybe `instance_name` is enough, because AFAIK the instance name has to be unique.

Reply:
But if it's there as a 'safety measure' then probably OK.

Reply:
I have removed the `plugin_name` argument. It's not needed, and now the function is consistent with others like `instance_get_devices`.